sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -158,12 +158,14 @@ case class CreateViewCommand(
       def verify(child: LogicalPlan) {
         child.collect {
           // Disallow creating permanent views based on temporary views.
-          case UnresolvedRelation(AsTableIdentifier(ident))
-            if sparkSession.sessionState.catalog.isTemporaryTable(ident) =>
-            // temporary views are only stored in the session catalog
-            throw new AnalysisException(s"Not allowed to create a permanent view $name by " +
-              s"referencing a temporary view $ident. " +
-              "Please create a temp view instead by CREATE TEMP VIEW")
+          case UnresolvedRelation(nameParts) if nameParts.length <= 2 =>
```

Comment:
It's orthogonal with this PR but I find the name parts with length comparison very confusing, in particular, ever since DSv2 added the catalog APIs with multi name parts.

Reply:
global temp view can have 2 name parts. The length check makes sure it's possible to be a temp view.